### PR TITLE
Drive development

### PR DESCRIPTION
**Pull request by miasya**
Created new modules as described in modules plan:

Interface CloudDocClient
class DriveClient implements CloudDocClient

Interface CloudSheet
class DriveCloudSheet implements CloudSheet

### CODE REVIEW

File: src/main/java/com/google/flourbot/api/DriveClient.java [hunk 1]
```diff
@@ -33,33 +30,34 @@
...
-  public void append();
-  public String createNew();
-}
+class DriveClient implements CloudDocClient {
 
-class MySheet implements DriveDoc {
+  static final String SPREADSHEET_SCOPE = "https://www.googleapis.com/auth/spreadsheets";
+
+  public CloudSheet getCloudSheet(String documentId) throws IOException, GeneralSecurityException {
+    DriveCloudSheet cs = new DriveCloudSheet(documentId, this.createService());
```
Comment (lipczak-g) [inline src/main/java/com/google/flourbot/api/DriveClient.java:39]: no need for the variable
return new DriveCloudSheet(documentId, this.createService());

File: src/main/java/com/google/flourbot/api/DriveClient.java [hunk 2]
```diff
@@ -33,33 +30,34 @@
...
-  // May not use this method
-  public String createNew() {
-    // make a new sheet and tell the user the ID and URL
-    Spreadsheet requestBody = new Spreadsheet()
-                  .setProperties(new SpreadsheetProperties()
-                    .setTitle("TEST"));;
-    Sheets.Spreadsheets.Create request = sheetsService.spreadsheets().create(requestBody);
-    Spreadsheet response = request.execute();
+    // Create sheetService
+    Sheets sheetsService = new Sheets.Builder(
```
Comment (lipczak-g) [inline src/main/java/com/google/flourbot/api/DriveClient.java:54]: same here, no need for the variable

File: src/main/java/com/google/flourbot/api/DriveCloudSheet.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  public DriveCloudSheet(String documentId, Sheets documentService) {
+    this.spreadsheetId = documentId;
+    this.sheetsService = documentService;
+  }
+
+  // TODO: replace hardcoded range with calculated range (so far values must be length 3)
+  public void appendRow(List<String> values) throws IOException, GeneralSecurityException {
+    
+    // Generate request to append to sheet
+    String range = "Sheet1!A1:C1";
```
Comment (lipczak-g) [inline src/main/java/com/google/flourbot/api/DriveCloudSheet.java:27]: not sure why the range is needed here when we can append lists of arbitrary length

File: src/main/java/com/google/flourbot/api/DriveCloudSheet.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    String valueInputOption = "USER_ENTERED";
```
Comment (lipczak-g) [inline src/main/java/com/google/flourbot/api/DriveCloudSheet.java:28]: these should be constants at the beginning of the class.

File: src/main/java/com/google/flourbot/api/DriveCloudSheet.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    Sheets.Spreadsheets.Values.Append request =
+      this.sheetsService.spreadsheets().values().append(this.spreadsheetId, range, requestBody);
+    request.setValueInputOption(valueInputOption);
+    request.setInsertDataOption(insertDataOption);
+    
+    AppendValuesResponse response = request.execute();
+  }
+  
+  // May not use this method
+  public String createNew() throws IOException, GeneralSecurityException {
```
Comment (lipczak-g) [inline src/main/java/com/google/flourbot/api/DriveCloudSheet.java:49]: delete if not used now, you can always find it in the history if we end up needing it.

**Comment by miasya** [comment]
@lipczak-g All suggested changes made in https://github.com/googleinterns/step81-2020/pull/24

**Pull request merged**